File: io_benchmark.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"time"
)

func main() {
	fmt.Println("Go IO Performance Test")
	filename := "testfile-go.txt"
	content := []byte("This is a line of text.\n")

	start := time.Now()

	// Create the file
	file, err := os.Create(filename)
	if err != nil {
		fmt.Println("Error creating file:", err)
		return
	}

	// Buffered writer is much more time-efficient!
	w := bufio.NewWriter(file)

	for i := 0; i < 1000000; i++ {
		_, err := w.Write(content)
		if err != nil {
			fmt.Println("Error writing to file:", err)
			return
		}
	}
	file.Close()

	writeDuration := time.Since(start)
	fmt.Printf("Write duration: %v\n", writeDuration)

	start = time.Now()

	// Read from file
	_, err = os.ReadFile(filename)
	if err != nil {
		fmt.Println("Error reading file:", err)
		return
	}

	readDuration := time.Since(start)
	fmt.Printf("Read duration: %v\n", readDuration)
}

File: IOBenchmark.java
import java.io.*;
import java.nio.file.*;

public class IOBenchmark {
    public static void main(String[] args) {
        System.out.printf("Java IO Performance Test%n");
        String filename = "testfile-java.txt";
        String content = "This is a line of text.\n";

        long start = System.currentTimeMillis();

        // Create the file
        FileWriter fileWriter;
        try {
            fileWriter = new FileWriter(filename);
        } catch (IOException e) {
            System.out.println("Error creating file: " + e.getMessage());
            return;
        }

        // Buffered writer is much more time-efficient!
        try (BufferedWriter writer = new BufferedWriter(fileWriter)) {
            for (int i = 0; i < 1000000; i++) {
                writer.write(content);
            }
        } catch (IOException e) {
            System.out.println("Error writing to file: " + e.getMessage());
            return;
        }

        long writeDuration = System.currentTimeMillis() - start;
        System.out.printf("Write duration: %d ms%n", writeDuration);

        start = System.currentTimeMillis();

        // Read from file
        try {
            byte[] fileContent = Files.readAllBytes(Paths.get(filename));
        } catch (IOException e) {
            System.out.println("Error reading file: " + e.getMessage());
            return;
        }

        long readDuration = System.currentTimeMillis() - start;
        System.out.printf("Read duration: %d ms%n", readDuration);
    }
}
